src/solve.jl

```diff
@@ -1,98 +1,112 @@
-function __solve(
-  prob::AbstractRODEProblem,
-alg::algType,timeseries=[],ts=[],ks=[],recompile::Type{Val{recompile_flag}}=Val{true};
-kwargs...) where {algType<:Union{AbstractRODEAlgorithm,AbstractSDEAlgorithm},recompile_flag}
-
-  integrator = __init(prob,alg,timeseries,ts,ks,recompile;kwargs...)
+function DiffEqBase.__solve(prob::DiffEqBase.AbstractRODEProblem,
+                            alg::Union{AbstractRODEAlgorithm,AbstractSDEAlgorithm},
+                            timeseries=[],ts=[],
+                            recompile::Type{Val{recompile_flag}}=Val{true};
+                            kwargs...) where recompile_flag
+  integrator = DiffEqBase.__init(prob,alg,timeseries,ts,recompile;kwargs...)
   solve!(integrator)
   integrator.sol
 end
 
-function __init(
-  prob::AbstractRODEProblem{uType,tType,isinplace,ND},
-  alg::algType,timeseries_init=uType[],ts_init=eltype(tType)[],ks_init=[],
+function DiffEqBase.__init(
+  prob::DiffEqBase.AbstractRODEProblem,
+  alg::Union{AbstractRODEAlgorithm,AbstractSDEAlgorithm},timeseries_init=typeof(prob.u0)[],
+  ts_init=eltype(prob.tspan)[],
   recompile::Type{Val{recompile_flag}}=Val{true};
-  dt = eltype(tType)(0),
-  timeseries_steps::Int = 1,
-  saveat = eltype(tType)[],tstops = eltype(tType)[],d_discontinuities= eltype(tType)[],
-  save_timeseries = nothing,
+  timeseries_steps = 1,
+  saveat = eltype(prob.tspan)[],
+  tstops = eltype(prob.tspan)[],
+  d_discontinuities= eltype(prob.tspan)[],
+  save_idxs = nothing,
   save_everystep = isempty(saveat),
   save_noise = save_everystep && typeof(prob.f) <: Tuple ?
-               has_analytic(prob.f[1]) : has_analytic(prob.f),
-  save_idxs = nothing,
+               DiffEqBase.has_analytic(prob.f[1]) : DiffEqBase.has_analytic(prob.f),
   save_on = true,
   save_start = save_everystep || isempty(saveat) || typeof(saveat) <: Number ? true : prob.tspan[1] in saveat,
   save_end = save_everystep || isempty(saveat) || typeof(saveat) <: Number ? true : prob.tspan[2] in saveat,
+  callback=nothing,
   dense = save_everystep && isempty(saveat),
   calck = (!isempty(setdiff(saveat,tstops)) || dense),
-  adaptive=isadaptive(alg),gamma=9//10,
-  abstol=nothing,reltol=nothing,
+  dt = eltype(prob.tspan)(0),
+  adaptive = isadaptive(alg),
+  gamma=9//10,
+  abstol=nothing,
+  reltol=nothing,
   qmax=qmax_default(alg),qmin=qmin_default(alg),
-  failfactor = 2,
   qoldinit=1//10^4, fullnormalize=true,
+  failfactor = 2,
   beta2=beta2_default(alg),
   beta1=beta1_default(alg,beta2),
-  delta=1//6,maxiters = 1e9,
-  dtmax=eltype(tType)((prob.tspan[end]-prob.tspan[1])),
-  dtmin=eltype(tType) <: AbstractFloat ? eps(eltype(tType)) : eltype(tType)(1//10^(10)),
-  internalnorm=ODE_DEFAULT_NORM,
-  unstable_check = ODE_DEFAULT_UNSTABLE_CHECK,
+  delta=1//6,
+  maxiters = 1000000,
+  dtmax=eltype(prob.tspan)((prob.tspan[end]-prob.tspan[1])),
+  dtmin = typeof(one(eltype(prob.tspan))) <: AbstractFloat ? 10*eps(eltype(prob.tspan)) :
```

Comment:
eps seems to work better

Reply:
Oh I've missed this change in OrdinaryDiffEq

Reply:
yeah, the reason is 10eps(Float32) > 1e-6 which is the minimum starting dt the auto-dt algorithm gives, so you get instant "below mindt" errors on some problems if you're using Float32s. While dropping the 10 doesn't seem to actually effect Float64 code, since if you're at dt=1e-15 you tend to go unstable to dt->0 anyways.